src/lib/markets/tx.ts

```diff
@@ -298,49 +189,43 @@ export async function buildSubmitVaaTx({ chainId, marketAddress, marketOracles,
 
 export type CancelOrderDetails = { market: Address; nonce: bigint } | OpenOrder
 
-export type BuildPlaceOrderTxArgs = {
-  pythClient: EvmPriceServiceConnection
+type BuildTriggerOrderBaseArgs = {
   address: Address
-  marketOracles?: MarketOracles
   marketAddress: Address
-  marketSnapshots?: MarketSnapshots
-  orderType: OrderTypes
-  limitPrice?: bigint
-  stopLoss?: bigint
-  takeProfit?: bigint
   side: PositionSide
-  collateralDelta?: bigint
   delta: bigint
-  positionAbs: bigint
   selectedLimitComparison?: TriggerComparison
-  referralFeeRate?: ReferrerInterfaceFeeInfo
-  interfaceFeeRate?: InterfaceFeeBps
-  cancelOrderDetails?: CancelOrderDetails[]
+  pythClient: EvmPriceServiceConnection
+  marketOracles?: MarketOracles
+  marketSnapshots?: MarketSnapshots
+  maxFee?: bigint
+  interfaceFee?: InterfaceFee
+  referralFee?: InterfaceFee
   onCommitmentError?: () => any
 } & WithChainIdAndPublicClient
 
-export async function buildPlaceOrderTx({
+export type BuildLimitOrderTxArgs = {
+  limitPrice: bigint
+  collateralDelta?: bigint
```

Comment:
We should remove collateral delta from here (and the corresponding update in the function itself) since they should just use an updateMarket tx to handle that.